tests/server/test_desired_state_versions.py

```diff
@@ -20,17 +20,112 @@
 import json
 import uuid
 from operator import itemgetter
+from typing import Any, Optional
 
 import pytest
 from tornado.httpclient import AsyncHTTPClient, HTTPRequest
 
 from inmanta import data
-from inmanta.data.model import DesiredStateLabel, PromoteTriggerMethod
-from inmanta.server import config
+from inmanta.agent.config import agent_reconnect_delay, server_timeout
+from inmanta.const import AGENT_SCHEDULER_ID
+from inmanta.data import ConfigurationModel, Scheduler
+from inmanta.data.model import DesiredStateLabel
+from inmanta.protocol import SessionEndpoint, methods
+from inmanta.server import SLICE_AGENT_MANAGER, config, protocol
+from inmanta.types import Apireturn
+from utils import retry_limited
+
+
+class BaseAgent(SessionEndpoint):
```

Comment:
`BaseAgent` sounds like this is an abstract base class. It also doesn't really say anything about what the class does or why it exists. I think we have to come up with a better name.